### PR TITLE
Added validation checks for interactive and non-interactive mode

### PR DESCRIPTION
**Pull request by dymurray**
Added sanity checks to ensure that the hostnames and IP addresses entered are in a valid format and on the correct subnets when necessary. Also added a check to see if the nameserver is an IP that is reachable.

### CODE REVIEW

**Comment by johnkim76** [comment]
ACK

File: bin/fusor-undercloud-configurator [hunk 1]
```diff
@@ -8,6 +8,48 @@ def getHwAddr(ifname):
   info = fcntl.ioctl(s.fileno(), 0x8927,  struct.pack('256s', ifname[:15]))
   return ':'.join(['%02x' % ord(char) for char in info[18:24]])
 
+def isInvalidIP(ip):
```
Comment (jmrodri) [inline bin/fusor-undercloud-configurator:11]: pep8 wants the methods to be is_invalid_ip vs camelCase.

File: bin/fusor-undercloud-configurator [hunk 2]
```diff
@@ -8,6 +8,48 @@ def getHwAddr(ifname):
...
+  if re.match('^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$', ip):
+    return False
+  return True
```
Comment (jmrodri) [inline bin/fusor-undercloud-configurator:14]: Why isn't this 
`return not re.match('^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$', ip)`

If it matches return false, if it doesn't match return true. By negating the match you turn this into a one line function.

File: bin/fusor-undercloud-configurator [hunk 3]
```diff
@@ -8,6 +8,48 @@ def getHwAddr(ifname):
...
+    return True
+  return False
+
+def validateAll(ip_addr, netmask, undercloud_ip, gateway):
+  if isInvalidIP(ip_addr):
+    print "Specified IP range is not valid"
+    os._exit(1)
+  if isInvalidNetmask(netmask):
+    print "Specified Netmask is not valid"
+    os._exit(1)
```
Comment (jmrodri) [inline bin/fusor-undercloud-configurator:39]: Should be sys.exit(1) not os._exit(1).

**Pull request merged**